test/spec/util/util.js

```diff
@@ -225,4 +225,32 @@ describe('iD.util', function() {
             expect(iD.utilUnicodeCharsTruncated('😎😬😆😵😴😄🙂🤔', 255)).to.eql('😎😬😆😵😴😄🙂🤔');
         });
     });
+
+    describe('utilNormalizeDateString', function() {
+        it('pads dates', function() {
+            expect(iD.utilNormalizeDateString('1970-01-01').value).to.eql('1970-01-01');
+            expect(iD.utilNormalizeDateString('1970-01-1').value).to.eql('1970-01-01');
+            expect(iD.utilNormalizeDateString('1970-1-01').value).to.eql('1970-01-01');
+            expect(iD.utilNormalizeDateString('-1-1-01').value).to.eql('-0001-01-01');
+            expect(iD.utilNormalizeDateString('123').value).to.eql('0123');
+            expect(iD.utilNormalizeDateString('-4003').value).to.eql('-4003'); // beyond displayable year range but still valid
+            expect(iD.utilNormalizeDateString('31337').value).to.eql('31337');
+            expect(iD.utilNormalizeDateString('-31337').value).to.eql('-31337');
```

Comment:
These years are well beyond the minimum year of 3999&nbsp;BCE set in OpenHistoricalMap/issues#28 and the maximum year equivalent to the current year at any given time. However, these year thresholds are only for display and filtering purposes. A truly primordial natural feature shouldn’t prevent the user from saving, nor should a building slated for demolition in a year’s time.